bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/SamsungTvAppWatchService.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.samsungtv.internal;
+
+import static java.nio.file.StandardWatchEventKinds.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchEvent.Kind;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.samsungtv.internal.protocol.RemoteControllerWebSocket;
+import org.openhab.core.OpenHAB;
+import org.openhab.core.service.AbstractWatchService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SamsungTvAppWatchService} provides a list of apps for >2020 Samsung TV's
+ * File should be in json format
```

Comment:
The code simply reads the file into a String array, ignoring comments. `remoteControllerWebSocket.updateAppList(allLines);` parses the JSON.

Reply:
Thanks, now followed that method. So each line is passed (except if starting with `#`) to that method and then parsed individually as JSON. To provide context for any other reviewers, from README:

```
# comment
{ "name":"Internet"                , "appId":"3202010022079"    , "type":2 }
{ "name":"Netflix"                 , "appId":"3201907018807"    , "type":2 }
```

Why this approach instead of using proper JSON for the complete file contents, so the file would essentially be a JSON file? Like this:

```
[
	{
		"name": "Internet",
		"appId": "3202010022079",
		"type": 2
	},
	{
		"name": "Netflix",
		"appId": "3201907018807",
		"type": 2
	}
]
```

This would be way faster to parse.

Reply:
Because the parser uses the existing `App` class. I don't have a TV that supports the app API, so I didn't want to mess with the existing app parsing. I designed the file format to fit with the existing app parser.

As the file is fairly short, and only gets loaded once, speed of parsing wasn't a concern.